src/api/mod.rs

```diff
@@ -403,6 +438,168 @@ impl<'a> ValueListBuilder<'a> {
     }
 }
 
+fn to_meta_data<'a, T>(meta_hm: T) -> Result<*mut meta_data_t, SubmitError>
```

Comment:
Since the string slice data should outlive the reference to the string slice data, a more correct representation is:
```rust
fn to_meta_data<'a, 'b: 'a, T>(meta_hm: T) -> Result<*mut meta_data_t, SubmitError>
where
    T: IntoIterator<Item = (&'a &'b str, &'a MetaValue)>,
```

But since it's all internal, it doesn't matter (and in fact I'm slightly regretting sending you down the generic route as I don't want to complicate the code too much if it's all internal). Up to you, leave it, convert it, or revert back to the hashmap reference -- it is all ok 👍

Reply:
i've done the lifetime modification. The declaration is certainly more complex with generic and all, but it did not modify the implementation part. So i think its a good trade-off between complexity and genericity.